### PR TITLE
Show extra errors for login

### PR DESCRIPTION
**Pull request by alex-deaconu**
## Description
Show extra errors for login

Network and Core internal errors handled
Consistent error parsing

Some cleanup

[stage-2]

## Motivation and Context
Allow users to review and fix network issues.
Better error messaging for Signup/Signin

## How Has This Been Tested?
Tested locally using Chrome's Dev tools and Charles proxy

## Release Plan:
- As the Submitter, upon requesting review of this pull request, I confirm that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed. 
- As the Reviewer, upon approving the changes in this PR, I confirm I have reviewed and I agree that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed

#### Release Checklist Items Skipped?
No

### CODE REVIEW

**Comment by alex-deaconu** [comment]
@ezequielc please review. Thanks!

File: web/scripts/components/userstate/controllers/ctr-login.js [hunk 1]
```diff
@@ -19,6 +19,30 @@ angular.module('risevision.common.components.userstate')
...
+          $scope.errors.messageTitle = messageTitle;
+          $scope.errors.message = message;
+        } else if (e.status === -1 || error.code === -1 || error.code === 0) {
+          $scope.errors.messageTitle = 'Hmm, we can\'t sign you ' + actionName +
+            ' because there\'s a problem with your connectivity.';
+          $scope.errors.message = 'Please check your connection and proxy or firewall settings and try again.';
+        } else {
+          $scope.errors.messageTitle = messageTitle;
+          $scope.errors.message = message;
+        }
```
Comment (ezequielc) [inline web/scripts/components/userstate/controllers/ctr-login.js]: @alex-deaconu I would suggest simplifying the logic here. It is a bit hard to follow and 
```
$scope.errors.messageTitle = messageTitle;
$scope.errors.message = message;
```
repeats in multiple cases.
Maybe something like:
```
if (e && (e.status >= 400 && e.status < 500)) {
...
else if (e && (e.status === -1 ...) {
...
else {
  $scope.errors.messageTitle = messageTitle;
  $scope.errors.message = message;
}
```
Thanks

Comment (alex-deaconu) [reply web/scripts/components/userstate/controllers/ctr-login.js]: @ezequielc originally I had generic messages for `(!e)` and the `else` clause. But then I added the message/messageTitle for all those cases as well.

I considered doing the refactor that you are suggesting, but I felt it was losing visibility for all the specific cases. However, I do agree it makes the code more readable.

Updated with a comment to reflect that. Please see the last commit. Thanks!

Comment (ezequielc) [reply web/scripts/components/userstate/controllers/ctr-login.js]: @alex-deaconu Sounds good. Thanks for updating. Looks good 👍
